### ISSUE TITLE
BUG: concat on axis=0 with categorical

### ISSUE THREAD

**Issue by jreback** [pydata/pandas#10177]
xref https://github.com/pydata/pandas/issues/9426, fixed in #9597 
``axis=1`` is ok

```
df1 = pandas.DataFrame(numpy.random.randn(6, 3), columns=["a", "b", "c"])
df2 = pandas.DataFrame(numpy.random.randn(7, 4), columns=["g", "h", "a", "c"])
df2['h'] = pandas.Series(pandas.Categorical(["one", "one", "two", "one", "two", "two", "one"]))
df = pandas.concat((df1, df2))
```

**Comment by sebp** [comment]
Note that it works if the "h" column is of object dtype:

```
          a         b         c         g    h
0 -0.673360 -0.795002  0.842495       NaN  NaN
1 -0.387503  1.093774 -1.138618       NaN  NaN
2 -1.108967 -0.339614  0.563358       NaN  NaN
3 -0.540412  1.299025  0.783089       NaN  NaN
4  1.796500 -1.630790 -1.039589       NaN  NaN
5 -0.427310 -0.149733  1.796212       NaN  NaN
0  0.087274       NaN  0.773546 -0.342498  one
1  0.346405       NaN -0.066082 -0.002910  one
2  1.496320       NaN  0.939827 -1.046683  two
3  3.248297       NaN -0.883009 -2.142893  one
4  1.315608       NaN  1.048771 -1.284582  two
5 -0.785210       NaN  0.158105  0.814332  two
6  0.322257       NaN -0.352677 -0.385390  one
```

**Comment by jreback** [comment]
@sebp want to give a shot at a fix?

**Comment by sebp** [comment]
@jreback sure, I can have a look at it if I find some time.

**Comment by jreback** [comment]
closed by #10179
